src/engine.js

```diff
@@ -47,27 +47,46 @@ class Engine extends EventEmitter {
     } else {
       if (!Object.prototype.hasOwnProperty.call(properties, 'event')) throw new Error('Engine: addRule() argument requires "event" property')
       if (!Object.prototype.hasOwnProperty.call(properties, 'conditions')) throw new Error('Engine: addRule() argument requires "conditions" property')
-
       rule = new Rule(properties)
     }
     rule.setEngine(this)
-
+    if (!rule.id) rule.id = '_' + Math.random().toString(36).substr(2, 9)
```

Comment:
This should probable be moved to rule.js, i.e. where you've added:
```js
if (options && (options.id && id !== 0)) {
	this.setId(options.id)
} else { // add this "else":
  this.setId('_' + Math.random().toString(36).substr(2, 9))
}
```

that way all rules default to having a generated id.

What do you think? Will that work?

Reply:
Oh, and we need a unit test for the auto-generated id

Reply:
@CacheControl you are right, i moved it to the Rule.js and added unit test as well